### PR TITLE
Remove uses of visitNodes and visitNode in visitEachChild

### LINKED ISSUE #49310: Error in 'setTextPos' during 'getCodeFixes'
This issue comes from crash dumps in telemetry. We've tried to de-duplicate issues on a best-effort basis, comparing the sequence of methods called and the command requested while ignoring line numbers.
**TypeScript version prefix**: 4.7.2
**VSCode version**: 1.68.0-insider
**Command requested**: getCodeFixes
**Hitting sessions**: 847
**Five line hash**: 8bf0402bbe40ec04bbb0073872731305
**Stack**:
```
    at setTextPos (tsserver.js:12102:22)
    at skipToEndOf (tsserver.js:145368:25)
    at processChildNodes (tsserver.js:146863:47)
    at <anonymous> (tsserver.js:146780:21)
    at visitNodes (tsserver.js:30468:24)
    at forEachChild (tsserver.js:30659:21)
    at processNode (tsserver.js:146777:20)
    at processChildNode (tsserver.js:146848:21)
    at <anonymous> (tsserver.js:146778:21)
    at visitNode (tsserver.js:30463:24)
    at forEachChild (tsserver.js:30622:21)
    at processNode (tsserver.js:146777:20)
    at formatSpanWorker (tsserver.js:146577:17)
    at <anonymous> (tsserver.js:146540:140)
    at getFormattingScanner (tsserver.js:145167:23)
    at formatNodeGivenIndentation (tsserver.js:146540:31)
    at getFormattedTextOfNode (tsserver.js:148971:45)
    at format (tsserver.js:148942:52)
    at computeNewText (tsserver.js:148945:23)
    at <anonymous> (tsserver.js:148909:39)
    at mapDefined (tsserver.js:605:30)
    at <anonymous> (tsserver.js:148907:42)
    at mapDefined (tsserver.js:605:30)
    at getTextChangesFromChanges (tsserver.js:148893:27)
    at ChangeTracker.getChanges (tsserver.js:148832:45)
    at ChangeTracker.with (tsserver.js:148252:32)
    at getCodeActions (tsserver.js:153962:68)
    at <anonymous> (tsserver.js:149499:77)
    at flatMap (tsserver.js:508:25)
    at getFixes (tsserver.js:149499:23)
    at <anonymous> (tsserver.js:164899:35)
    at flatMap (tsserver.js:508:25)
    at getCodeFixesAtPosition (tsserver.js:164897:23)
    at unknown (suppressed.js:182:34)
    at unknown (suppressed.js:79:25)
    at Session.getCodeFixes (tsserver.js:176796:64)
    at _a.<computed> (tsserver.js:175179:61)
    at <anonymous> (tsserver.js:177109:88)
    at Session.executeWithRequestId (tsserver.js:177100:28)
    at Session.executeCommand (tsserver.js:177109:33)
    at Session.onMessage (tsserver.js:177135:35)
    at <anonymous> (tsserver.js:179770:31)
```

### PR DESCRIPTION
**Pull request by jakebailey**
Fixes #49310

### CODE REVIEW

**Review by sheetalkamat — APPROVED** [review]

**Comment by andrewbranch** [comment]
I don’t understand why this is needed. Don’t synthesized nodes emitted by the TextChanges writer get their `pos` and `end` set?

**Comment by jakebailey** [comment]
Not some, it seems. I'll go figure out who's making this array and why it's not getting fixed, then.

**Comment by andrewbranch** [comment]
It’s still quite possible this is a good fix, I’d just like to understand it a little more.

**Comment by jakebailey** [comment]
Aha, you were good to wonder. There's a number of long-standing typos in `visitEachChild` where `visitNodes` was called instead of `nodesVisitor`, or `visitNode` instead of `nodeVisitor`. Wow!

File: src/compiler/visitorPublic.ts [hunk 1]
```diff
@@ -616,7 +616,7 @@ namespace ts {
                     nodeVisitor(node.argument, visitor, isTypeNode),
                     nodeVisitor(node.assertions, visitor, isNode),
                     nodeVisitor(node.qualifier, visitor, isEntityName),
-                    visitNodes(node.typeArguments, visitor, isTypeNode),
+                    nodesVisitor(node.typeArguments, visitor, isTypeNode),
```
Comment (jakebailey) [inline src/compiler/visitorPublic.ts:619]: This one in particular was the bug.

I tried to come up with a way that would prevent this sort of from happening (like `Debug.type<unknown>(visitNodes)`, or redeclaring it), but wasn't successful in stopping TS from allowing this.

Comment (jakebailey) [reply src/compiler/visitorPublic.ts:619]: I really, really wonder how many other bugs there are that were caused by this in the wild...

**Review by andrewbranch — APPROVED** [review]
Haha wow. Glad I asked.

**Review by DanielRosenwasser — APPROVED** [review]

File: src/compiler/visitorPublic.ts (hunk 1 above)
Comment (jakebailey) [reply src/compiler/visitorPublic.ts:619]: >  wasn't successful in stopping TS from allowing this.

The fix would be to start targeting ES2015 aka ES6, then do `const visitNodes = undefined` in the body; the only reason that doesn't work now is that we downlevel default parameters to assignments and that interferes with the scoping.

**Pull request merged**